### PR TITLE
[web] use typed SVG API

### PR DESCRIPTION
**Pull request by yjbanov**
- Use typed SVG API instead of SVG strings for compatibility with [TrustedTypes](https://w3c.github.io/webappsec-trusted-types/dist/spec/).
- Remove `NullTreeSanitizer`.

Fixes b/198428641

### CODE REVIEW

**Review by hterkelsen — APPROVED** [review]
LGTM

File: lib/web_ui/lib/src/engine/html/dom_canvas.dart [hunk 1]
```diff
@@ -273,31 +273,33 @@ String _borderStrokeToCssUnit(double value) {
...
           paint.strokeWidth != null)) {
-    sb.write('stroke="${colorToCssString(color)}" ');
-    sb.write('stroke-width="${paint.strokeWidth ?? 1.0}" ');
-    sb.write('fill="none" ');
+    svgPath.setAttribute('stroke', '${colorToCssString(color)}');
+    svgPath.setAttribute('stroke-width', '${paint.strokeWidth ?? 1.0}');
+    svgPath.setAttribute('fill', 'none');
   } else if (paint.color != null) {
-    sb.write('fill="${colorToCssString(color)}" ');
+    svgPath.setAttribute('fill', '${colorToCssString(color)}');
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/html/dom_canvas.dart:296]:
```suggestion
    svgPath.setAttribute('fill', colorToCssString(color));
```

File: lib/web_ui/lib/src/engine/html/dom_canvas.dart [hunk 2]
```diff
@@ -273,31 +273,33 @@ String _borderStrokeToCssUnit(double value) {
...
+  root.append(svgPath);
   final ui.Color color = paint.color ?? const ui.Color(0xFF000000);
   if (paint.style == ui.PaintingStyle.stroke ||
       (paint.style != ui.PaintingStyle.fill &&
           paint.strokeWidth != 0 &&
           paint.strokeWidth != null)) {
-    sb.write('stroke="${colorToCssString(color)}" ');
-    sb.write('stroke-width="${paint.strokeWidth ?? 1.0}" ');
-    sb.write('fill="none" ');
+    svgPath.setAttribute('stroke', '${colorToCssString(color)}');
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/html/dom_canvas.dart:292]:
```suggestion
    svgPath.setAttribute('stroke', colorToCssString(color));
```

**Review by mdebbar — APPROVED** [review]

File: lib/web_ui/lib/src/engine/html/bitmap_canvas.dart [hunk 3]
```diff
@@ -1170,6 +1172,140 @@ String? stringForBlendMode(ui.BlendMode? blendMode) {
...
+const int SVG_FEBLEND_MODE_HARD_LIGHT = 9;
+const int SVG_FEBLEND_MODE_SOFT_LIGHT = 10;
+const int SVG_FEBLEND_MODE_DIFFERENCE = 11;
+const int SVG_FEBLEND_MODE_EXCLUSION = 12;
+const int SVG_FEBLEND_MODE_HUE = 13;
+const int SVG_FEBLEND_MODE_SATURATION = 14;
+const int SVG_FEBLEND_MODE_COLOR = 15;
+const int SVG_FEBLEND_MODE_LUMINOSITY = 16;
+
+// Source: https://osscs.corp.google.com/chromium/chromium/src/+/main:third_party/blink/renderer/platform/graphics/graphics_types.cc;l=55;drc=59bdabfb96e4c7a0de476b24c34ff197d1fa2ce4
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/html/bitmap_canvas.dart:1196]: Good catch! Done.

File: lib/web_ui/lib/src/engine/html/bitmap_canvas.dart [hunk 4]
```diff
@@ -1170,6 +1172,140 @@ String? stringForBlendMode(ui.BlendMode? blendMode) {
...
+/// Flutter's [BlendMode] flattens what SVG expresses as two orthogonal
+/// properties, a composite operator and blend mode. Instances of this class
+/// are returned from [blendModeToSvgEnum] by mapping Flutter's [BlendMode]
+/// enum onto the SVG equivalent.
+///
+/// See also:
+///
+///  * https://www.w3.org/TR/compositing-1
+///  * https://osscs.corp.google.com/chromium/chromium/src/+/main:third_party/blink/renderer/platform/graphics/graphics_types.cc;l=55;drc=59bdabfb96e4c7a0de476b24c34ff197d1fa2ce4
+///  * https://osscs.corp.google.com/chromium/chromium/src/+/main:third_party/blink/renderer/modules/canvas/canvas2d/base_rendering_context_2d.cc;l=725;drc=59bdabfb96e4c7a0de476b24c34ff197d1fa2ce4
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/html/bitmap_canvas.dart:1221]: Done.

File: lib/web_ui/lib/src/engine/html/dom_canvas.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/html/dom_canvas.dart:296]: Done.

File: lib/web_ui/lib/src/engine/html/dom_canvas.dart (hunk 2 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/html/dom_canvas.dart:292]: Done.

**Pull request merged**